src/scale/Time.ts

```diff
@@ -206,8 +206,10 @@ class TimeScale extends IntervalScale<TimeScaleSetting> {
         let upperUnitIndex = primaryTimeUnits.length - 1;
         let maxLevel = 0;
         each(ticks, tick => {
-            upperUnitIndex = Math.min(upperUnitIndex, indexOf(primaryTimeUnits, tick.time.upperTimeUnit));
-            maxLevel = Math.max(maxLevel, tick.time.level);
+            if (tick.time) {
```

Comment:
I noticed that TypeScript suppresses this TypeError as per the current tsconfig.json, where the `strictNullChecks` option is disabled. But if we enable it, there are many errors to be fixed. For simplicity, I agree with your current fix by adding the safeguard.

@100pah How do you think about this?